### PR TITLE
Added Lorentz force module and unit tests

### PR DESCRIPTION
**Pull request by jkblc**
Added Lorentz force module using functional programming and unit tests using assert

### CODE REVIEW

**Comment by jkblc** [comment]
The pylint error is due to the assignment of a lambda expression to a variable, which pylint considers unnecessary but it is important for the assignment.

File: lorentz_force/lorentz_unit_tests.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    Test case where the angle between velocity and magnetic field is 90 degrees (π/2 radians).
+    The sin(90 degrees) is 1, so the force should be maximized.
+    """
+    charge = 1.6e-19  # Charge in Coulombs
+    velocity = 2e6    # Velocity in m/s
+    magnetic_field = 0.01  # Magnetic field in Tesla
+    angle_radians = math.pi / 2  # 90 degrees
+
+    expected_force = charge * velocity * magnetic_field  # sin(90 degrees) is 1
+    assert math.isclose(lorentz_force(
```
Comment (dnxjay) [inline lorentz_force/lorentz_unit_tests.py:19]: Missing tolerance in math.isclose()

File: lorentz_force/lorentz_unit_tests.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+"""
+Module to test lorentz_force.py
+"""
+
+import math
+from lorentz_force import lorentz_force
+
+def test_lorentz_force_at_90_degrees():
```
Comment (dnxjay) [inline lorentz_force/lorentz_unit_tests.py:8]: Potential for floating-point precision issues in test_lorentz_force_at_0_degrees()

File: lorentz_force/lorentz_unit_tests.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+"""
```
Comment (dnxjay) [inline lorentz_force/lorentz_unit_tests.py:1]: Consider adding tests for edge cases or more general situations such as:
When charge is zero.
When velocity is zero.
When magnetic field is zero.
Negative charges

**Review by dnxjay — CHANGES_REQUESTED** [review]
Docstrings missing and address the testing issues by adding more cases

File: lorentz_force/lorentz_unit_tests.py (hunk 1 above)
Comment (jkblc) [reply lorentz_force/lorentz_unit_tests.py:19]: What should the tolerance be? By default it uses a tolerance of 1e-9

Comment (dnxjay) [reply lorentz_force/lorentz_unit_tests.py:19]: upon more research the default tolerance of 1e-9 is generally sufficient for many computational physics problems

**Comment by jkblc** [comment]
Doc strings are not typically used for lambda functions. Should it be refactored into a function?

File: lorentz_force/lorentz_force.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+in a magnetic field using functional programming concepts.
+
+It demonstrates the use of lambda functions and the map function to calculate
+the Lorentz force for a list of particles.
+"""
+
+import math
+
+# Lambda function to calculate Lorentz force
+lorentz_force = lambda charge, velocity, magnetic_field, angle_radians: (
```
[the remark this answers is not in the document]
Comment (jkblc) [reply lorentz_force/lorentz_force.py:12]: Makes sense, I will fix it.

**Comment by laserlab** [comment]
looks good now merging
